straxen/scada.py

```diff
@@ -238,75 +215,57 @@ def _query_single_parameter(self,
         if not isinstance(every_nth_value, int):
             raise ValueError('"value_every_seconds" must be an int!')
 
-        # First we have to create an array where we can fill values with
-        # the sampling frequency of scada:
-        seconds = np.arange(start, end + 1, 10**9)  # +1 to make sure endtime is included
+        # First we have to create an array where we can fill values:
+        seconds = np.arange(start, end+1, 10**9)  # +1 to make sure endtime is included
         df = pd.DataFrame()
         df.loc[:, 'time'] = seconds
         df['time'] = df['time'].astype('<M8[ns]')
         df.set_index('time', inplace=True)
 
-        # Check if first value is in requested range:
-        query = self.SCADA_SECRETS.copy()
+        # Init parameter query:
+        query = self.SCADA_SECRETS.copy()  # Have to copy dict since it is not immutable
         query['name'] = parameter_name
-        query['EndDateUnix'] = (start // 10**9) + 1  # +1 since it is end before exclusive
-        query = urllib.parse.urlencode(query)
-        values = requests.get(self.SCLastValue_URL + query)
 
-        try:
-            temp_df = pd.read_json(values.text)
-        except ValueError as e:
-            mes = values.text
-            query_message = ast.literal_eval(mes)
-            raise ValueError(f'SCADA raised a value error when looking for '
-                             f'your parameter "{parameter_name}". The error '
-                             f'was {query_message}') from e
+        # Check if first value is in requested range:
+        temp_df = self._query(query,
+                              self.SCLastValue_URL,
+                              end=(start // 10**9) + 1)  # +1 since it is end before exclusive
 
         # Store value as first value in our df
         df.loc[df.index.values[0], parameter_key] = temp_df['value'][0]
 
-        # Query values between start+1 and end time:
-        query = self.SCADA_SECRETS.copy()
-        query["StartDateUnix"] = (start // 10**9) + 1
-        query["EndDateUnix"] = (end // 10**9)
-        query['name'] = parameter_name
-        query = urllib.parse.urlencode(query)
-        values = requests.get(self.SCData_URL + query)
-
-        try:
-            # Here we cannot do any better since the Error message returned
-            # by the scada api is always the same...
-            temp_df = pd.read_json(values.text)
-            self._raw_data = temp_df
-            df.loc[temp_df['timestampseconds'], parameter_key] = temp_df.loc[:, 'value'].values
-        except ValueError:
-            pass
-
-        # Let user decided whether to ffill or interpolate:
-        if interpolation:
+        # Query values between start+1 and endtime:
+        offset = 0
+        ntries = 0
+        max_tries = 40000  # This corresponds to ~23 years
+        while ntries < max_tries:
```

Comment:
at this stage, you probably don't need 1s resolution anymore, does it make sense to pack this in too (at some point)?

Reply:
I added a new feature which allows to query longer time ranges. There seems to be an issue with the WebAPI so this is why I deactivated `query_type_lab` manually for the moment. Depending on if I receive an answer by Hagar we can include it either in the upcoming or next release.